### ISSUE TITLE
Accessibility for dropdowns

### ISSUE THREAD

**Issue by jennur** [inveniosoftware/invenio-app-rdm#1363]
Improvements are needed for the accessibility in dropdowns. Make sure they are working well with the keyboard and that the screenreader is announcing the content correctly.

See example https://www.w3.org/TR/wai-aria-practices/examples/combobox/combobox-select-only.html

## Global account dropdown
Accessible by keyboard, but not announced well by the screenreader. Tested with VoiceOver on Mac.

<img width="303" alt="Screenshot 2022-03-08 at 10 14 11" src="https://user-images.githubusercontent.com/21052053/157206451-c2f28aa6-fc63-4fbf-987d-ce6846a8a611.png">

## Pagination
Accessible by keyboard, but selects the option when navigating. It should be possible to navigate the options and select manually.

<img width="182" alt="Screenshot 2022-03-08 at 10 14 19" src="https://user-images.githubusercontent.com/21052053/157206462-7629e54b-a8ad-4478-899a-a435acf99ecb.png">

## Sort results
Same situation as for pagination.

<img width="263" alt="Screenshot 2022-03-08 at 10 14 30" src="https://user-images.githubusercontent.com/21052053/157206472-f438de11-9789-40b2-96a0-3e92f8bab375.png">
